components/browser/state/src/main/java/mozilla/components/browser/state/action/BrowserAction.kt

```diff
@@ -152,6 +152,20 @@ sealed class TabListAction : BrowserAction() {
      */
     data class AddMultipleTabsAction(val tabs: List<TabSessionState>) : TabListAction()
 
+    /**
+     * Moves a set of tabIDs into a new position (maintaining internal ordering)
+     *
+     * @property tabIds The IDs of the tabs to move.
+     * @property position The position in the list to move them to.
+     * @property tabsFilter Apply a filter to the tabs to offset the [position].
+     * Needed to deal with the inactive tabs not being in the main tab tray.
+     */
+    data class MoveTabsAction(
+        val tabIds: List<String>,
+        val position: Int,
+        val tabsFilter: (TabSessionState) -> Boolean = { true }
+    ) : TabListAction()
```

Comment:
After discussing this with @csadilek, we were thinking that the API would be simpler to understand if we split this into two actions, one for normal tabs and one for private tabs. This way we do not need to provide `tabsFilter`, which is way more generic than needed. After all we want to either move in the list of private or normal tabs.

Reply:
That will fail if there are any inactive tabs (right now Fenix is split between Normal/active, Inactive, and Private), and Inactive tabs' definition seems like it should be changeable based on user preference (right now it's hardcoded 4 days)

What do you think of the other API I proposed (target a specific tab, place the list before/after it based on a Bool)?

Reply:
Ah, I see. It's biting us that "inactive tabs" are not actually in a separate bucket in `BrowserState` and that on this site here we have no clue that the list of displayed tabs is only a subset. In hindsight it would have been nice to have this "inactive filter" / "displayed tabs" filter inside the state, so that we can use it.

> What do you think of the other API I proposed (target a specific tab, place the list before/after it based on a Bool)?

That's a great idea. Replacing the index with a specific tab may help. The before/after flag is needed for the edge cases start/end of list, right? It could also be a more evolved state modeled with a sealed class, which could have children like `AtBeginning`, `AfterTab(id)`, `AtEnd`.

Reply:
Correct, before/after is needed for the start/end of the list. It could also be useful for if tabs are moved and then a deactivated tab is reactivated. If the tab list is AbCD, b deactivated, then moving D after A results in ADbC whereas before C would be AbDC).